### ISSUE TITLE
padded-blocks: add an option: maxStatements

### ISSUE THREAD

**Issue by fab1o** [eslint/eslint#12997]
**What rule do you want to change?**

[padded-blocks](https://eslint.org/docs/rules/padded-blocks)

**Does this change cause the rule to produce more or fewer warnings?**

Fewer warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option called `maxStatements` would only warn if the block has more than x statements. (name can be changed)

Something like:

```js
padded-blocks: ["error", "always", { maxStatements: 1 }]
```

**Please provide some example code that this change will affect:**

👍 Examples of **correct** code for this rule with the `"always", { maxStatements: 1 }"` options:
```js
/*eslint padded-blocks: ["error", "always", { maxStatements: 1 }]*/
if (a) {
    b();
}

if (a)
{

    b();
    c();

}

if (a) {
    // comment
    b();
}
```

👎 Examples of **incorrect** code for this rule with the `"always", { maxStatements: 1 }"` options:

```js
if (a) {
    b(); c();
}

if (a) { b(); }

if (a)
{

    b();

}

if (a) {
    b();

}

if (a) {
    // comment
    b();

}
```

**What does the rule currently do for this code?**

It forces the blocks to be padded with an x amount of lines.

**What will the rule do after it's changed?**

Allows that blocks with an x amount of lines not be padded.

**Are you willing to submit a pull request to implement this change?**

**Comment by anikethsaha** [comment]
+1 for this !
also, I guess options name `offset` or `maxoffset` would fit better !

cc @kaicataldo @mdjermanovic  !

**Comment by kaicataldo** [comment]
Can you add examples for how this would interact with the other options, such as `"never"` and `"allowSingleLineBlocks"`? I'm also curious how this would interact with the individual `"blocks|classes|switches"` options.

**Comment by anikethsaha** [comment]
as per what I understood from it !

- `never` : it wont effect any as `never` doesnt allow padding !

- `allowSingleLineBlocks` : I am not sure about its operation as it should report for [`this example right ?`](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgcGFkZGVkLWJsb2NrczogW1wiZXJyb3JcIiwgXCJhbHdheXNcIiwgeyBhbGxvd1NpbmdsZUxpbmVCbG9ja3M6IHRydWUgfV0qL1xuaWYgKGEpIHtcbiAgXG4gIHZhciBhYSA9IGFcbiAgYigpO1xuICBcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibm8tdXNlLWJlZm9yZS1kZWZpbmUiOjIsInBhZGRlZC1ibG9ja3MiOjJ9LCJlbnYiOnt9fX0=)

- for the rest of the options (`"blocks|classes|switches"`), it would be a second check. counting the top and bottom lines around the first/last node of blockstatements nd counting the numbers of line, it it exceeds this limit, report !

**Comment by kaicataldo** [comment]
> * never : it wont effect any as never doesnt allow padding !

I don't think this is accurate. The proposal states:
> New option called maxStatements would only warn if the block has more than x statements.

Regarding the other options, I think it's valuable to see how this configuration would look and what it would actually catch as part of the evaluation process. Since all the options will interact with each other, it feels like this rule would get exponentially more complicated.

**Comment by anikethsaha** [comment]
> I don't think this is accurate. The proposal states:

> > New option called maxStatements would only warn if the block has more than x statements.


Well, in that case, I don't think it should be applied when its `never` as `never` means not to have padding and we cant set max limit where the limit is already 0

> Since all the options will interact with each other, it feels like this rule would get exponentially more complicated.

I do see a point of more options here but I think its kind of important for keeping the code style more consistent and uniform.  As I mentioned, it should be a second check and default as none so there should not be any check for this unless its being passed in the option

**Comment by kaicataldo** [comment]
I do think it's important to make sure we evaluate how all these options interact. I'm not currently convinced the added complexity is worth it.
